source/class/qx/tool/cli/Application.js

```diff
@@ -29,11 +29,14 @@ qx.Class.define("qx.tool.cli.Application", {
      * during startup of the application
      */
     async main() {
+      process.exitCode = 0;
       try {
         await new qx.tool.cli.Cli().run();
       } catch (e) {
         qx.tool.compiler.Console.error("Error: " + (e.stack || e.message));
-        process.exit(1);
+        process.exitCode = -1;
```

Comment:
Failure code is 1, didn't find -1 value in nodejs docs

Reply:
I missed this before. There are a few issues here. Firstly, this `process.exitCode = -1;` will not even be executed. `process.exit(1)` is the last line of JS code that will be executed.

Exit codes are, by definition in some environments, and by convention in others, 8-bit unsigned values, with 0 meaning success and 1-255 meaning various types of error exits. Using negative values as exit codes is typically discouraged, and in most environments, simply maps to the unsigned equivalent. `process.exit(-1)`, for example, will usually yield an exit code of `255` as seen by the parent process.

`process.exitCode` is typically used when there are multiple places that could fail. You can set `process.exitCode` to a unique value for each of those failure locations, and then do `process.exit()` (with no argument) to exit with the current value of `process.exitCode`